src/main/java/org/reactome/release/qa/common/QACheckerHelper.java

```diff
@@ -273,192 +278,43 @@ public static String getAttributeTableName(String clsName,
-				physicalEntities.addAll(findAllConstituentPEs(physicalEntity));
-				// If EWAS, just adds to Set and is returned.
-			} else if (physicalEntity.getSchemClass().isa(ReactomeJavaConstants.EntityWithAccessionedSequence)) {
-				physicalEntities.add(physicalEntity);
-			}
-		}
-		return physicalEntities;
-	}
-
-	/**
-	 * Checks if PhysicalEntity contains additional PhysicalEntities within it.
-	 * @param physicalEntity GKInstance -- PhysicalEntity that is being checked.
-	 * @return boolean -- true if PhysicalEntity type that contains multiple PEs, false if only type that has single PhysicalEntity.
-	 */
-	public static boolean containsMultiplePEs(GKInstance physicalEntity) {
-		return physicalEntity.getSchemClass().isa(ReactomeJavaConstants.Complex) ||
-				physicalEntity.getSchemClass().isa(ReactomeJavaConstants.Polymer) ||
-				physicalEntity.getSchemClass().isa(ReactomeJavaConstants.EntitySet);
-	}
-
-	/**
-	 * Finds all PhysicalEntities contained within a Complex, Polymer or EntitySet. Searches recursively by calling parent 'findAllPhysicalEntities' method.
-	 * @param multiPEInstance GKInstance -- Complex, Polymer or EntitySet instance that will be searched for all PhysicalEntity instances.
-	 * @return Set<GKInstance> -- All distinct PhysicalEntities that are found in Complex/Polymer/EntitySet.
-	 * @throws Exception -- Thrown by MySQLAdaptor.
-	 */
-	public static Set<GKInstance> findAllConstituentPEs(GKInstance multiPEInstance) throws Exception {
-		Set<GKInstance> physicalEntities = new HashSet<>();
-		if (multiPEInstance.getSchemClass().isa(ReactomeJavaConstants.Complex) || multiPEInstance.getSchemClass().isa(ReactomeJavaConstants.Polymer)) {
-			// Recursively search Complex/Polymer for all PhysicalEntities.
-			for (GKInstance constituentPE : getComplexOrPolymerConstituentPEs(multiPEInstance)) {
-				physicalEntities.addAll(findAllPhysicalEntities(constituentPE));
-			}
-		} else if (multiPEInstance.getSchemClass().isa(ReactomeJavaConstants.EntitySet)) {
-			// Recursively search members and candidates (if EntitySet is a CandidateSet) for all PhysicalEntities.
-			physicalEntities.addAll(findEntitySetPhysicalEntities(multiPEInstance));
-		}
-		return physicalEntities;
-	}
-
-	/**
-	 * Helper method that returns either a Complexes' components or Polymers' repeatedUnits.
-	 * @param multiPEInstance GKInstance -- Either a Complex or Polymer instance.
-	 * @return Set<GKInstance> -- All components or repeatedUnits from the incoming instance.
+	 * This method utilizes the InstanceUtilities method for finding all reaction participants. It also checks for the
+	 * 'activeUnit' participant in catalysts/regulations that can be flagged by certain QA checks. Generally speaking,
+	 * the 'activeUnit' instance should be found in the 'physicalEntity' of catalysts/regulations. Sometimes this is not
+	 * the case, when the wrong activeUnit is mistakenly checked in. This will be flagged by QA tests that call this method.
```

Comment:
Just curious - If this method returns all participants, how is this method used by other methods to determine if the activeUnit specifically is wrong?

Reply:
As we discussed yesterday (but I'm posting here for posterity) -- the method isn't meant to specifically target activeUnits and check that they match with their catalyst/regulation physicalEntity, but rather the nature of some of the QA tests that call this method is to flag PhysicalEntity instance types (such as would be found in the activeUnit attribute) that have a differing species from a Reaction they are associated with.